ingress: Set interval annotation if omitted (#42)
containeroo/cloudflare-operator: docs/content/core_concept.md

   If you change the `IP` object, cloudflare-operator will automatically update the `DNSRecord.spec.content` fields.

   Example:

   ```yaml
   apiVersion: cf.containeroo.ch/v1alpha1
   kind: IP
   metadata:
     name: static-address
   spec:
     type: static
     address: 142.251.36.35
   ```

2. **Dynamic DNS**

   If the `type` is set to `dynamic`, cloudflare-operator will fetch your external IPv4 address in a specified interval (`spec.interval`).

   Example:

   ```yaml
   apiVersion: cf.containeroo.ch/v1alpha1
   kind: IP
   metadata:
     name: external-ipv4
   spec:
     type: dynamic
     interval: 5m
   ```
   
   If no `dynamicIPSources` are specified, cloudflare-operator will use a hardcoded set of sources.  
   If you prefer other sources, you can add them as a list in `dynamicIPSources`.
   
   Example:
   
   ```yaml
   apiVersion: cf.containeroo.ch/v1alpha1
   kind: IP
   metadata:
     name: external-ipv4
   spec:
     type: dynamic
     dynamicIPSources:
       - https://api.ipify.org
     interval: 5m
   ```

>    

Reviewer: ```suggestion

```

   !!! warning
       The source must return only the external IPv4 address.

>    

Reviewer: ```suggestion

```

       Good example:

>    

Reviewer: ```suggestion

```

       ```bash
       curl https://api.ipify.org
       ```

>    

Reviewer: ```suggestion

```

       Output:

>    

Reviewer: ```suggestion

```

       ```console
       142.251.36.35
       ```

>    

Reviewer: ```suggestion

```

       Bad example:

>    

Reviewer: ```suggestion

```

       ```bash
       curl "https://api.ipify.org?format=json"
       ```

>    

Reviewer: ```suggestion

```

       Output:

>    

Reviewer: ```suggestion

```

       ```console
       {"ip":"142.251.36.35"}
       ```

>    

Reviewer: ```suggestion

```

| `cf.containeroo.ch/ttl`      | `1` or `60`-`86400`    | Time to live, in seconds, of the Cloudflare DNS record. Must be between 60 and 86400, or 1 for 'automatic'                                  |
| `cf.containeroo.ch/type`     | `A` or `CNAME`         | Cloudflare DNS record type                                                                                                                  |
| `cf.containeroo.ch/interval` | `5m`                   | Interval at which cloudflare-operator will compare the Cloudflare DNS record with the `DNSRecord` object and reconcile the DNSRecord object |
| `cf.containeroo.ch/ignore`   | `true` or `false`      | Skip creation of a DNS record                                                                                                               |

!!! note "cf.containeroo.ch/ignore"
    If you add the label `cf.containeroo.ch/ignore=true` and cloudflare-operator has already created a `DNSRecord`, cloudflare-operator will clean up the `DNSRecord` (Kubernetes and Cloudflare).

!!! note "cf.containeroo.ch/interval"

>     If you delete the `DNSRecord` object, cloudflare-operator will automatically recreate it with the `interval` specified in the annotation.

Author: ```suggestion
    If you delete the `DNSRecord` object, cloudflare-operator will automatically recreate at the next `Ingress` reconciliation loop. If not specified, the default interval `5m` will be used.
```
